--- doxygen-1.7.3.orig/src/memberdef.cpp	2011-01-02 14:17:45.000000000 +0100
+++ doxygen-1.7.3.orig/src/memberdef.cpp	2011-03-23 16:47:58.490450171 +0100
@@ -101,6 +101,7 @@ 
   //ol.disableAllBut(OutputGenerator::Html);
   bool htmlOn  = ol.isEnabled(OutputGenerator::Html);
   bool latexOn = ol.isEnabled(OutputGenerator::Latex);
+  bool manOn = ol.isEnabled(OutputGenerator::Man);
   {
     // html and latex
     if (htmlOn)  ol.enable(OutputGenerator::Html);
@@ -108,7 +109,8 @@ 
     ol.endMemberDocName();
     ol.startParameterList(!md->isObjCMethod()); 
   }
-  ol.enableAll();
+  ol.popGeneratorState();
+  ol.pushGeneratorState();
   ol.disable(OutputGenerator::Html);
   ol.disable(OutputGenerator::Latex);
   {
@@ -210,13 +212,13 @@ 
       ol.disable(OutputGenerator::Man);
       ol.disable(OutputGenerator::Latex);
       ol.startEmphasis();
-      ol.enable(OutputGenerator::Man);
+      if (manOn) ol.enable(OutputGenerator::Man);
       if (latexOn) ol.enable(OutputGenerator::Latex);
       if (a->name.isEmpty()) ol.docify(a->type); else ol.docify(a->name);
       ol.disable(OutputGenerator::Man);
       ol.disable(OutputGenerator::Latex);
       ol.endEmphasis();
-      ol.enable(OutputGenerator::Man);
+      if (manOn) ol.enable(OutputGenerator::Man);
       if (latexOn) ol.enable(OutputGenerator::Latex);
     }
     if (!a->array.isEmpty())
@@ -273,7 +275,6 @@ 
   ol.disable(OutputGenerator::Html);
   ol.disable(OutputGenerator::Latex);
   if (!md->isObjCMethod()) ol.docify(")"); // end argument list
-  ol.enableAll();
   if (htmlOn) ol.enable(OutputGenerator::Html);
   if (latexOn) ol.enable(OutputGenerator::Latex);
   if (first) ol.startParameterName(defArgList->count()<2);
--- doxygen-1.7.3.orig/src/outputgen.cpp	2010-05-09 14:49:09.000000000 +0200
+++ doxygen-1.7.3.orig/src/outputgen.cpp	2011-03-23 15:53:44.768583075 +0100
@@ -58,6 +58,7 @@ 
 
 void OutputGenerator::endPlainFile()
 {
+  t.unsetDevice();
   delete file;
   file=0;
   fileName.resize(0);
