diff -up doxygen-1.8.12/addon/doxywizard/doxywizard.rc doxygen-1.8.12/addon/doxywizard/doxywizard
diff -up doxygen-1.8.12/doc/translator.py.rc doxygen-1.8.12/doc/translator.py
--- doxygen-1.8.12/doc/translator.py.rc	2016-05-10 13:51:51.000000000 -0600
+++ doxygen-1.8.12/doc/translator.py	2016-12-21 22:34:41.007132208 -0700
@@ -83,7 +83,7 @@ def xopen(fname, mode='r', encoding='utf
     the default 'utf-8-sig' is used (skips the BOM automatically).
     '''
 
-    major, minor, patch = (int(e) for e in platform.python_version_tuple())
+    major, minor, patch = (int(re.sub('[^0-9].*','',e)) for e in platform.python_version_tuple())
     if major == 2:
         return open(fname, mode=mode) # Python 2 without encoding
     else:
@@ -1990,7 +1990,7 @@ class TrManager:
 if __name__ == '__main__':
 
     # The Python 2.6+ or 3.3+ is required.
-    major, minor, patch = (int(e) for e in platform.python_version_tuple())
+    major, minor, patch = (int(re.sub('[^0-9].*','',e)) for e in platform.python_version_tuple())
     if (major == 2 and minor < 6) or (major == 3 and minor < 0):
         print('Python 2.6+ or Python 3.0+ are required for the script')
         sys.exit(1)
