diff -up doxygen-1.5.8/configure.orig doxygen-1.5.8/configure
--- doxygen-1.5.8/configure.orig	2008-12-26 20:22:39.000000000 +0100
+++ doxygen-1.5.8/configure	2009-02-06 11:00:44.000000000 +0100
@@ -485,6 +485,7 @@ INSTTOOL  = $f_insttool
 DOXYDOCS  = ..
 DOCDIR    = $f_docdir
 QTDIR     = $QTDIR
+MAN1DIR    = share/man/man1
 EOF
 
 if test "$f_dot" != NO; then
diff -up doxygen-1.5.8/addon/doxywizard/Makefile.in.orig doxygen-1.5.8/addon/doxywizard/Makefile.in
--- doxygen-1.5.8/addon/doxywizard/Makefile.in.orig	2009-02-06 14:55:30.000000000 +0100
+++ doxygen-1.5.8/addon/doxywizard/Makefile.in	2009-02-06 14:56:08.000000000 +0100
@@ -29,11 +29,11 @@ distclean: Makefile.doxywizard
 	$(RM) Makefile.doxywizard
 
 install:
-	$(INSTTOOL) -d $(INSTALL)/bin	
-	$(INSTTOOL) -m 755 ../../bin/doxywizard $(INSTALL)/bin	
-	$(INSTTOOL) -d $(INSTALL)/$(MAN1DIR)
+	$(INSTTOOL) -d $(DESTDIR)$(INSTALL)/bin	
+	$(INSTTOOL) -m 755 ../../bin/doxywizard $(DESTDIR)$(INSTALL)/bin	
+	$(INSTTOOL) -d $(DESTDIR)$(INSTALL)/$(MAN1DIR)
 	cat ../../doc/doxywizard.1 | sed -e "s/DATE/$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > doxywizard.1
-	$(INSTTOOL) -m 644 doxywizard.1 $(INSTALL)/$(MAN1DIR)/doxywizard.1
+	$(INSTTOOL) -m 644 doxywizard.1 $(DESTDIR)$(INSTALL)/$(MAN1DIR)/doxywizard.1
 	rm doxywizard.1
 
 FORCE:
diff -up doxygen-1.5.8/Makefile.in.orig doxygen-1.5.8/Makefile.in
--- doxygen-1.5.8/Makefile.in.orig	2009-02-06 15:00:41.000000000 +0100
+++ doxygen-1.5.8/Makefile.in	2009-02-06 15:00:54.000000000 +0100
@@ -44,8 +44,6 @@ distclean: clean
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
-
 install: doxywizard_install 
 	$(INSTTOOL) -d $(DESTDIR)/$(INSTALL)/bin
 	$(INSTTOOL) -m 755 bin/doxygen    $(DESTDIR)/$(INSTALL)/bin
diff -up doxygen-1.5.8/tmake/lib/linux-g++/tmake.conf.orig doxygen-1.5.8/tmake/lib/linux-g++/tmake.conf
--- doxygen-1.5.8/tmake/lib/linux-g++/tmake.conf.orig	2009-02-06 17:08:45.000000000 +0100
+++ doxygen-1.5.8/tmake/lib/linux-g++/tmake.conf	2009-02-06 17:12:16.000000000 +0100
@@ -11,7 +11,7 @@ TMAKE_CC		= gcc
 TMAKE_CFLAGS		= -pipe
 TMAKE_CFLAGS_WARN_ON	= -Wall -W -fno-exceptions
 TMAKE_CFLAGS_WARN_OFF	=
-TMAKE_CFLAGS_RELEASE	= -O2
+TMAKE_CFLAGS_RELEASE	= $(RPM_OPT_FLAGS)
 TMAKE_CFLAGS_DEBUG	= -g
 TMAKE_CFLAGS_SHLIB	= -fPIC
 TMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
@@ -27,12 +27,12 @@ TMAKE_CXXFLAGS_YACC	= $$TMAKE_CFLAGS_YAC
 
 TMAKE_INCDIR		=
 TMAKE_LIBDIR		=
-TMAKE_INCDIR_X11	= /usr/X11R6/include
-TMAKE_LIBDIR_X11	= /usr/X11R6/lib
-TMAKE_INCDIR_QT		= $(QTDIR)/include
-TMAKE_LIBDIR_QT		= $(QTDIR)/lib
-TMAKE_INCDIR_OPENGL	= /usr/X11R6/include
-TMAKE_LIBDIR_OPENGL	= /usr/X11R6/lib
+TMAKE_INCDIR_X11	=
+TMAKE_LIBDIR_X11	=
+TMAKE_INCDIR_QT		=
+TMAKE_LIBDIR_QT		=
+TMAKE_INCDIR_OPENGL	=
+TMAKE_LIBDIR_OPENGL	=
 
 TMAKE_LINK		= g++
 TMAKE_LINK_SHLIB	= g++
diff -up doxygen-1.5.8/tmake/lib/macosx-uni-c++/tmake.conf.orig doxygen-1.5.8/tmake/lib/macosx-uni-c++/tmake.conf
