diff -up doxygen-1.8.11/src/util.cpp.than doxygen-1.8.11/src/util.cpp
--- doxygen-1.8.11/src/util.cpp.than	2016-03-06 11:13:21.831912181 +0100
+++ doxygen-1.8.11/src/util.cpp	2016-03-06 11:15:20.359280738 +0100
@@ -6732,6 +6732,9 @@ QCString latexEscapePDFString(const char
       case '\\': t << "\\textbackslash{}"; break;
       case '{':  t << "\\{"; break;
       case '}':  t << "\\}"; break;
+      case '_': t << "\\_"; break;
+      case '%':  t << "\\%"; break;
+      case '&':  t << "\\&"; break;
       default:
         t << c;
         break;
