diff --git a/src/classdef.cpp b/src/classdef.cpp
index 51fb8c6..111d59b 100644
--- a/src/classdef.cpp
+++ b/src/classdef.cpp
@@ -573,7 +573,9 @@ void ClassDef::internalInsertMember(MemberDef *md,
                   break;
                 case Public:
                   addMemberToList(MemberListType_pubTypes,md,TRUE);
-                  isSimple=QCString(md->typeString()).find(")(")==-1;
+                  isSimple=!md->isEnumerate() &&
+                           !md->isEnumValue() &&
+                           QCString(md->typeString()).find(")(")==-1; // func ptr typedef
                   break;
                 case Private:
                   addMemberToList(MemberListType_priTypes,md,TRUE);
