diff -up doxygen-1.5.6/addon/doxywizard/Makefile.in.redhat doxygen-1.5.6/addon/doxywizard/Makefile.in
--- doxygen-1.5.6/addon/doxywizard/Makefile.in.redhat	2008-05-19 11:28:20.000000000 +0200
+++ doxygen-1.5.6/addon/doxywizard/Makefile.in	2008-05-19 11:28:42.000000000 +0200
@@ -41,7 +41,7 @@ clean: Makefile.doxywizard 
 distclean: clean
 	$(RM) Makefile.doxywizard version.cpp config.l config.h portable.h portable.cpp
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install:
 	$(INSTTOOL) -d $(INSTALL)/bin	
diff -up doxygen-1.5.6/Makefile.in.redhat doxygen-1.5.6/Makefile.in
--- doxygen-1.5.6/Makefile.in.redhat	2008-04-20 09:59:24.000000000 +0200
+++ doxygen-1.5.6/Makefile.in	2008-05-19 11:18:09.000000000 +0200
@@ -50,7 +47,7 @@ distclean: clean
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install: doxywizard_install 
 	$(INSTTOOL) -d $(INSTALL)/bin
diff -up doxygen-1.5.6/tmake/lib/linux-g++/tmake.conf.redhat doxygen-1.5.6/tmake/lib/linux-g++/tmake.conf
--- doxygen-1.5.6/tmake/lib/linux-g++/tmake.conf.redhat	2006-07-21 21:54:13.000000000 +0200
+++ doxygen-1.5.6/tmake/lib/linux-g++/tmake.conf	2008-05-19 11:18:09.000000000 +0200
@@ -11,7 +11,7 @@ TMAKE_CC		= gcc
 TMAKE_CFLAGS		= -pipe
 TMAKE_CFLAGS_WARN_ON	= -Wall -W -fno-exceptions
 TMAKE_CFLAGS_WARN_OFF	=
-TMAKE_CFLAGS_RELEASE	= -O2
+TMAKE_CFLAGS_RELEASE	= $(RPM_OPT_FLAGS)
 TMAKE_CFLAGS_DEBUG	= -g
 TMAKE_CFLAGS_SHLIB	= -fPIC
 TMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
@@ -50,7 +50,7 @@ TMAKE_LIBS_QT_MT	= -lqt-mt
 TMAKE_LIBS_QT_OPENGL	= -lqgl
 TMAKE_LIBS_OPENGL	= -lMesaGL -lMesaGLU -lXmu
 
-TMAKE_MOC		= moc
+TMAKE_MOC		= $(QTDIR)/bin/moc
 
 TMAKE_AR		= ar cqs
 TMAKE_RANLIB		=
